Deploying.md

```diff
@@ -129,3 +130,160 @@ There is a crontab that syncs the switchboard. If you have changed something in
+         #
+         # Put the following in /etc/uwsgi/apps-available/{app-name}.ini:
+         #
+         #    [uwsgi]
+         #    ini = /opt/backoffice/{app-name}/uwsgi.ini
+         #
+         # and make a symlink to it from /etc/uwsgi/apps-enabled/{app-name}.ini.
+         #
+         # It is assumed that Pipenv is configured to install its virtualenvs in .venv
+         # with PIPENV_VENV_IN_PROJECT=1.
+         #
+         [uwsgi]
+         plugin = python3
+         envdir = %d/env.d/uwsgi
+         virtualenv = /opt/{app-name}/.venv
+         wsgi-file = /opt/{app-name}/app.py
+         processes = 1
+         threads = 2
+         enable-threads = true
+         ```
+         See the [husky-musher uwsgi.ini file](https://github.com/seattleflu/backoffice/blob/master/husky-musher/uwsgi.ini) as an example.
+         You may want to choose a different number of processes and threads than what is used in the example above.
+     * A new envdir diretory at `env.d/uwsgi`.
+     Inside it, add the non-sensitive environment variables.
+
+   * **systemd workflow:**
+     * A `Makefile` that contains:
+         ```make
+         SHELL := /bin/bash -euo pipefail
+
+         apps := \
+            {app-name}.service
+
+         install: $(apps:%=/etc/systemd/system/%)
+
+         /etc/systemd/system/%: %
+            @install -cv $< $@
+         ```
+         See the [scan-switchboard Makefile](https://github.com/seattleflu/backoffice/blob/master/scan-switchboard/Makefile) as an example.
+     * A systemd service file named `{app-name}.service` that contains:
+         ```ini
+         [Unit]
+         Description=SCAN Switchboard
+         After=network.target
+
+         [Service]
+         WorkingDirectory=/opt/{app-name}
+         User=nobody
+         Environment="PIPENV_VENV_IN_PROJECT=1"
+         ExecStart=/usr/local/bin/pipenv run ./bin/serve --config base_url:/{desired-endpoint}/
+         Restart=always
+         RestartSec=10
+
+         [Install]
+         WantedBy=default.target
+         ```
+
+6. Update [backoffice] repo documentation README pointing to the newly created directory from the previous step.
+7. Deploy the [backoffice] repo changes.
+   1. From the `/opt/backoffice` directory on the backoffice server, run `git pull`.
+   2. Add any secret environment variables under `/opt/backoffice/{app-name}/env.d/`.
+8. Deploy your app for the first time.
+   With `{app-name}` as your desired application name:
+   * **uWSGI workflow:**
+     1. On the backoffice server, put the following in `/etc/uwsgi/apps-available/{app-name}.ini`:
+         ```ini
+         [uwsgi]
+         ini = /opt/backoffice/{app-name}/uwsgi.ini
+         ```
+         and make a symlink to it from `/etc/uwsgi/apps-enabled/{app-name}.ini`.
+     2. `sudo chgrp -R www-data env.d/uwsgi` TODO: Didn't catch all of this when pairing w/ Tom
+     3. `chmod -R g=nXo = env.d/uwsgi` TODO: Didn't catch all of this when pairing with Tom
```

Comment:
lingering TODOs

Reply:
FYI @tsibley I would love your eyes on the additional steps we did in the uWSGI first-time deployment of the husky-musher. I didn't completely catch everything.

Reply:
I got it now :)